File: 1502.py
file = open('_inputs_/1502.1')
a = []
for line in file:
    a.append([int(_) for _ in line.strip().split('x')])
# a = [[2,3,4]]
res = 0
for l, w, h in a:
    # print(l, w, h, min(l, w, h))
    res += 2*l*w + 2*w*h + 2*l*h + min(l*w, w*h, l*h)
print(res)
res2 = 0
for l, w, h in a:
    bow = l*w*h
    wp = (l + w + h - max(l, w, h)) * 2
    res2 += bow + wp
print(res2)

File: 1502.cc
#include "iostream"
#include "cstdio"

using	namespace std;

int	m3(int a, int b, int c, int);

int	main()
{
	string s;
	int a, b, c;
	int res = 0;
	int res2 = 0;
	while (!cin.eof() && cin >> s)
	{
		sscanf(s.c_str(), "%dx%dx%d", & a, & b, & c);
		res += 2*a*b + 2*b*c + 2*a*c + m3(a*b, b*c, a*c,0);
		int wp = a*b*c;
		int bow = (a + b + c - m3(a, b, c,1)) * 2;
		res2 += wp + bow;
	}
	cout << res << '\n' << res2 << endl;
}

int	m3(int a, int b, int c, int mode)
{
	int r;
	if (mode)
	{
		r = a > b ? a : b;
		r = r > c ? r : c;
	}
	else
	{
		r = a < b ? a : b;
		r = r < c ? r : c;
	}
	return r;
}
